sdk/src/main/java/com/qonversion/android/sdk/storage/UserPropertiesStorage.kt

```diff
@@ -12,11 +20,30 @@ class UserPropertiesStorage : PropertiesStorage {
 
     override fun clear(properties: Map<String, String>) {
         properties.keys.map {
-            userProperties.remove(it)
+            if (userProperties[it] == properties[it]) {
+                userProperties.remove(it)
+            }
         }
     }
 
     override fun getProperties(): Map<String, String> {
         return userProperties.toMap()
     }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun getHandledProperties(): Map<String, String> {
+        return try {
+            val jsonString: String = preferences.getString(HANDLED_PROPERTIES_KEY + config.uid, null) ?: ""
```

Comment:
I would suggest here to concatenate strings into one variable as it's used in several places .
For example:
`val uniqueHandledPropertiesKey = "$HANDLED_PROPERTIES_KEY${config.uid}"
`